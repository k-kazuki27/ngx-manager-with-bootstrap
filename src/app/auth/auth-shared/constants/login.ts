export const LOGIN_ERROR = 'ユーザID、もしくはパスワードが間違っています。'
export const REQUIRED_ERROR = 'ユーザIDとパスワードを入力してください。'
