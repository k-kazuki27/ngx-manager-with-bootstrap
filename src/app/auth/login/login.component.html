<div class="login-page" [@routerTransition]>
  <div class="row justify-content-md-center">
    <div class="col-md-4">
      <img
        src="assets/images/login/logo.png"
        width="150px"
        class="user-avatar"
      />
      <h1>管理画面</h1>
      <form [formGroup]="form" autocomplete="off">
        <div class="form-content">
          <div class="form-group">
            <input
              type="text"
              class="form-control input-underline input-lg"
              name="userId"
              formControlName="userId"
              placeholder="ユーザID"
            />
          </div>
          <div class="form-group">
            <input
              type="password"
              class="form-control input-underline input-lg"
              name="password"
              formControlName="password"
              placeholder="パスワード"
            />
          </div>
          <p *ngIf="errorMessage" class="text-danger">
            <i class="fa fa-times-circle-o"></i>{{ errorMessage }}
          </p>
        </div>
        <a class="btn rounded-btn" (click)="login()">ログイン</a>
      </form>
    </div>
  </div>
</div>
