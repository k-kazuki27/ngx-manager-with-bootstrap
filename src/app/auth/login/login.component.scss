$topnav-background-color: #222;

:host {
  display: block;
}

.login-page {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 3em;
  overflow: auto;
  color: #fff;
  text-align: center;
  background: $topnav-background-color;

  .col-lg-4 {
    padding: 0;
  }

  .input-lg {
    height: 46px;
    padding: 10px 16px;
    font-size: 18px;
    line-height: 1.3333333;
    border-radius: 0;
  }

  .input-underline {
    color: #fff;
    background: 0 0;
    border: none;
    border-bottom: 2px solid rgba(255, 255, 255, 0.5);
    border-radius: 0;
    box-shadow: none;
  }

  .input-underline:focus {
    border-bottom: 2px solid #fff;
    box-shadow: none;
  }

  .rounded-btn {
    width: 100%;
    padding: 0 25px;
    color: rgba(255, 255, 255, 0.8);
    font-size: 18px;
    line-height: 40px;
    background: $topnav-background-color;
    border: 2px solid rgba(255, 255, 255, 0.8);
    -webkit-border-radius: 50px;
    border-radius: 50px;
  }

  .rounded-btn:hover,
  .rounded-btn:focus,
  .rounded-btn:active,
  .rounded-btn:visited {
    color: rgba(255, 255, 255, 1);
    border: 2px solid rgba(255, 255, 255, 1);
    outline: none;
  }

  h1 {
    margin-top: 20px;
    margin-bottom: 10px;
    font-weight: 300;
    font-size: 36px;

    small {
      color: rgba(255, 255, 255, 0.7);
    }
  }

  .form-group {
    padding: 8px 0;

    input::-webkit-input-placeholder {
      color: rgba(255, 255, 255, 0.6) !important;
    }

    input:-moz-placeholder {
      /* Firefox 18- */
      color: rgba(255, 255, 255, 0.6) !important;
    }

    input:-ms-input-placeholder {
      color: rgba(255, 255, 255, 0.6) !important;
    }
  }

  .form-content {
    padding: 40px 0;
  }

  .user-avatar {
    border: 2px solid #fff;
    -webkit-border-radius: 50%;
    border-radius: 50%;
  }
}
