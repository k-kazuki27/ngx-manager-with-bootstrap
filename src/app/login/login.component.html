<div class="login-page" [@routerTransition]>
  <div class="row justify-content-md-center">
    <div class="col-md-4">
      <img
        src="assets/images/login/logo.png"
        width="150px"
        class="user-avatar"
      />
      <h1>Sample App</h1>
      <form role="form">
        <div class="form-content">
          <div class="form-group">
            <input
              type="text"
              class="form-control input-underline input-lg"
              placeholder="ユーザID"
            />
          </div>
          <div class="form-group">
            <input
              type="password"
              class="form-control input-underline input-lg"
              placeholder="パスワード"
            />
          </div>
        </div>
        <a
          class="btn rounded-btn"
          [routerLink]="['/dashboard']"
          (click)="onLoggedin()"
          >ログイン</a
        >
      </form>
    </div>
  </div>
</div>
