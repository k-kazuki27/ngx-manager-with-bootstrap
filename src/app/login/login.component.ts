import { Component, OnDestroy, OnInit } from '@angular/core'
import { FormBuilder, FormControl, FormGroup } from '@angular/forms'
import { Router } from '@angular/router'
import { Subject } from 'rxjs'
import { takeUntil } from 'rxjs/operators'

import { routerTransition } from '../router.animations'
import { LoginApi, LoginUserService, StorageService } from '../shared'

const LOGIN_ERROR = 'ユーザID、もしくはパスワードが間違っています。'
const REQUIRED = 'ユーザIDとパスワードを入力してください。'

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss'],
  animations: [routerTransition()]
})
export class LoginComponent implements OnInit, OnDestroy {
  form: FormGroup
  errorMessage: string

  private onDestroy$ = new Subject()

  constructor(
    private router: Router,
    private fb: FormBuilder,
    private loginApi: LoginApi,
    private loginUserService: LoginUserService,
    private storageService: StorageService
  ) {}

  ngOnInit() {
    this.storageService.removeAuth()

    this.form = this.fb.group({
      userId: [null, []],
      password: [null, []]
    })
    this.errorMessage = null
  }

  ngOnDestroy() {
    this.onDestroy$.next()
  }

  get userId(): FormControl {
    return this.form.get('userId') as FormControl
  }
  get password(): FormControl {
    return this.form.get('password') as FormControl
  }

  login(): void {
    if (!this.userId.value || !this.password.value) {
      this.errorMessage = REQUIRED
      return
    }

    this.loginApi
      .login(
        {
          userId: this.userId.value,
          password: this.password.value
        },
        'body',
        true
      )
      .pipe(takeUntil(this.onDestroy$))
      .subscribe(
        res => {
          this.loginUserService.setLoginUser(res)
          this.storageService.setAuth()
          sessionStorage.setItem('isLoggedin', 'true')
          this.router.navigate(['/'])
        },
        () => {
          this.errorMessage = LOGIN_ERROR
        }
      )
  }
}
