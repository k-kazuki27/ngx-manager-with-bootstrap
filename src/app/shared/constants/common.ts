export const REG_EXP_HALF_ALPHA_NUM_SYMBOL = '^[a-zA-Z0-9!-/:-@Â¥[-`{-~]+$'
