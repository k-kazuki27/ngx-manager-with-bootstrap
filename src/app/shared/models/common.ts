export interface TableHeader {
  isButton: boolean
  isSort: boolean
  key?: string
  name?: string
}

export interface Page {
  pageNo: number
  itemsPerPage: number
}
