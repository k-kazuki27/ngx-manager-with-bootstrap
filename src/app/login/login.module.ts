import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'
import { ReactiveFormsModule } from '@angular/forms'

import { LoginRoutingModule } from './login-routing.module'
import { LoginComponent } from './login.component'

@NgModule({
  imports: [CommonModule, LoginRoutingModule, ReactiveFormsModule],
  declarations: [LoginComponent]
})
export class LoginModule {}
