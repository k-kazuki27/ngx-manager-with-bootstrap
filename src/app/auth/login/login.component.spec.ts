import { HttpClientModule } from '@angular/common/http'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
import { RouterTestingModule } from '@angular/router/testing'

import { AuthModule } from '../auth.module'
import { LoginComponent } from './login.component'

describe('LoginComponent', () => {
  let component: LoginComponent
  let fixture: ComponentFixture<LoginComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientModule,
        AuthModule,
        RouterTestingModule,
        BrowserAnimationsModule
      ]
    }).compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create', () => {
    expect(component).toBeTruthy()
    expect(fixture).toMatchSnapshot()
  })
})
