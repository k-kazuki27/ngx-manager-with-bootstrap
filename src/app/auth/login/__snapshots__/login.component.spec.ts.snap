// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should create 1`] = `
<app-login
  errorMessage="null"
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  loginApi={[Function LoginApi]}
  loginUserService={[Function LoginUserService]}
  onDestroy$={[Function Subject]}
  router={[Function Router]}
  storageService={[Function StorageService]}
>
  <div
    class="login-page ng-trigger ng-trigger-routerTransition"
    style=""
  >
    <div
      class="row justify-content-md-center"
    >
      <div
        class="col-md-4"
      >
        <img
          class="user-avatar"
          src="assets/images/login/logo.png"
          width="150px"
        />
        <h1
          class="ng-tns-c0-0"
        >
          管理画面
        </h1>
        <form
          autocomplete="off"
          class="ng-tns-c0-0 ng-untouched ng-pristine ng-valid"
          ng-reflect-form="[object Object]"
          novalidate=""
        >
          <div
            class="form-content"
          >
            <div
              class="form-group"
            >
              <input
                class="form-control input-underline input-lg ng-untouched ng-pristine ng-valid"
                formcontrolname="userId"
                name="userId"
                ng-reflect-name="userId"
                placeholder="ユーザID"
                type="text"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control input-underline input-lg ng-untouched ng-pristine ng-valid"
                formcontrolname="password"
                name="password"
                ng-reflect-name="password"
                placeholder="パスワード"
                type="password"
              />
            </div>
            
          </div>
          <a
            class="btn rounded-btn"
          >
            ログイン
          </a>
        </form>
      </div>
    </div>
  </div>
</app-login>
`;
